Cant compile 2.0.1 version in PHP 5.4.26

I have been compiled 1.0.1 version on my server, and its works good.

but, when i try to make latest version (2.0.1), it shows an error :

In file included from /usr/local/include/php/Zend/zend_compile.h:690,
from /usr/local/include/php/Zend/zend_modules.h:26,
from /usr/local/include/php/Zend/zend_API.h:26,
from /usr/local/include/php/main/php.h:38,
from ./src/pthreads.h:34,
from ./src/lock.h:33,
from /home/admin/pthreads-2.0.1/src/lock.c:26:
/usr/local/include/php/Zend/zend_vm_opcodes.h:168:7: warning: no newline at end of file
/home/admin/pthreads-2.0.1/src/lock.c: In function 'pthreads_lock_alloc':
/home/admin/pthreads-2.0.1/src/lock.c:45: error: 'PTHREAD_MUTEX_RECURSIVE_NP' undeclared (first use in this function)
/home/admin/pthreads-2.0.1/src/lock.c:45: error: (Each undeclared identifier is reported only once
/home/admin/pthreads-2.0.1/src/lock.c:45: error: for each function it appears in.)
**\* [src/lock.lo] Error code 1

I'm using FreeBSD 9.1 amd64 and PHP 5.4.26

--- a/src/lock.c
+++ b/src/lock.c
@@ -35,7 +35,7 @@ pthreads_lock pthreads_lock_alloc(TSRMLS_D) {
 
 		pthread_mutexattr_init(&attr);
 
-#ifdef PTHREAD_MUTEX_RECURSIVE
+#if defined(PTHREAD_MUTEX_RECURSIVE) || defined(__FreeBSD__)
 		pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
 #else
 		pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE_NP);

